features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/messagecomposer/MessageComposerPresenter.kt

```diff
@@ -212,13 +228,13 @@ class MessageComposerPresenter @Inject constructor(
     private fun CoroutineScope.sendAttachment(
         attachment: Attachment,
         attachmentState: MutableState<AttachmentsState>,
-    ) = launch {
-        when (attachment) {
-            is Attachment.Media -> {
+    ) = when (attachment) {
```

Comment:
There is some code duplication in the class `AttachmentsPreviewPresenter`, is there a (easy) way to reduce this?

Reply:
The 2 functions are quite similar, but they pass different data to their `sendMedia` counterparts. We could try wrapping those in  some use case or helper class, but I'm not sure it's worth it for just 2 occurrences. Also, I think we want to remove the upload progress from the `AttachmentsPreviewView` in the future as display it on the the timeline item instead, so part of this code will probably disappear.